workloads/files/workload-network-script-cm.yml

```diff
@@ -0,0 +1,500 @@
+      done
+      infra_nodes=`oc get nodes -l pbench_agent=true,node-role.kubernetes.io/infra= --no-headers | awk '{print $1}'`
+      for node in $infra_nodes; do
+        echo "infra" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+      worker_nodes=`oc get nodes -l pbench_agent=true,node-role.kubernetes.io/worker= --no-headers | awk '{print $1}'`
+      for node in $worker_nodes; do
+        echo "worker" > /var/lib/pbench-agent/tools-default/remote@$node
+      done
+    fi
+    source /opt/pbench-agent/profile
+    log "Done configuring pbench Network benchmarks"
+    # End pbench Configuration
+
+    # Start of Test Code
+    log "Running Network benchmarks"
+
+    config_prefix="P2P"
+    cp /root/workload/pod-rc-template.yaml /tmp/pod-rc-template.yaml
+    if [[ "${NETWORK_TEST_HOSTNETWORK}" == "true" ]]; then
+      config_prefix="P2P_HN"
+      envsubst < /root/workload/network-scc.yaml.template > /tmp/network-scc.yaml
+      oc replace --force -f /tmp/network-scc.yaml
+      cp /root/workload/pod-hn-rc-template.yaml /tmp/pod-rc-template.yaml
+    fi
+    # Loop over pair count
+    for pair_count in ${NETWORK_TEST_PAIRS//,/ }
+    do
+      log "Running Network benchmarks with ${pair_count} pair(s)"
+      if [ "${NETWORK_TEST_SERVICE}" == "true" ]; then
+        config_prefix="S2S"
+        PAIR_COUNT=${pair_count} envsubst < /root/workload/s2s-network.yaml.template > /tmp/network.yaml
+        cp /root/workload/svc-template.yaml /tmp/svc-template.yaml
+        # Append required ports to service template
+        for i in `seq ${NETWORK_TEST_UPERF_PORT_RANGE}`
+        do
+          echo -e "    - name: tcp-${i}\n      port: ${i}\n      protocol: TCP" >> /tmp/svc-template.yaml
+          echo -e "    - name: udp-${i}\n      port: ${i}\n      protocol: UDP" >> /tmp/svc-template.yaml
+        done
+      else
+        PAIR_COUNT=${pair_count} envsubst < /root/workload/p2p-network.yaml.template > /tmp/network.yaml
+      fi
+
+      VIPERCONFIG=/tmp/network.yaml openshift-tests run-test "[Feature:Performance][Serial][Slow] Load cluster should load the cluster [Suite:openshift]"
+
+      # Check if clients and servers are ready
+      for rc_num in `seq 0 $((${pair_count} - 1))`
+      do
+        retries=0
+        while [ $retries -le 60 ] ; do
+          client_ready=`oc get replicationcontroller/uperf-client-${rc_num} -o json -n network0 | jq '.status.replicas==.status.availableReplicas'`
+          if [ "${client_ready}" == "true" ]; then
+            log "uperf-client-${rc_num} is ready"
+            break
+          else
+            log "Waiting for uperf-client-${rc_num} to be ready"
+            sleep 2
+          fi
+          retries=$[$retries + 1]
+        done
+        if [ "${client_ready}" == "false" ]; then
+          log "uperf-client-${rc_num} failed to be ready on time"
+          exit 1
+        fi
+        retries=0
+        while [ $retries -le 60 ] ; do
```

Comment:
Perhaps move into a function?  Code dupliation of the entire client section.

Reply:
Got it, moved this and the log function into a separate functions.sh file sourced at the top.